velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -260,16 +278,65 @@ TEST_F(DateTimeFunctionsTest, dateAdd) {
   EXPECT_EQ(parseDate("1969-12-31"), dateAdd(parseDate("5881580-07-11"), kMin));
 }
 
-TEST_F(DateTimeFunctionsTest, dateSub) {
-  const auto dateSubFunc = [&](std::optional<int32_t> date,
-                               std::optional<int32_t> value) {
-    return evaluateOnce<int32_t, int32_t>(
-        "date_sub(c0, c1)", {date, value}, {DATE(), INTEGER()});
+TEST_F(DateTimeFunctionsTest, dateAddSmallint) {
+  const auto dateAdd = [&](std::optional<int32_t> date,
+                           std::optional<int16_t> value) {
+    return evaluateDateFuncOnce<int32_t, int16_t>(
+        "date_add(c0, c1)", date, value);
   };
 
+  // Check null behaviors
+  EXPECT_EQ(std::nullopt, dateAdd(std::nullopt, 1));
+  EXPECT_EQ(std::nullopt, dateAdd(parseDate("2019-02-28"), std::nullopt));
+  EXPECT_EQ(std::nullopt, dateAdd(std::nullopt, std::nullopt));
+
+  // Check simple tests.
+  EXPECT_EQ(parseDate("2019-03-01"), dateAdd(parseDate("2019-03-01"), 0));
+  EXPECT_EQ(parseDate("2019-03-01"), dateAdd(parseDate("2019-02-28"), 1));
+
+  // Account for the last day of a year-month
+  EXPECT_EQ(parseDate("2020-02-29"), dateAdd(parseDate("2019-01-30"), 395));
+  EXPECT_EQ(parseDate("2020-02-29"), dateAdd(parseDate("2019-01-30"), 395));
+
+  // Check for negative intervals
+  EXPECT_EQ(parseDate("2019-02-28"), dateAdd(parseDate("2020-02-29"), -366));
+  EXPECT_EQ(parseDate("2019-02-28"), dateAdd(parseDate("2020-02-29"), -366));
+
+  // Check for minimum and maximum tests.
+  EXPECT_EQ(
+      parseDate("2059-09-17"), dateAdd(parseDate("1969-12-31"), kMaxSmallint));
+  EXPECT_EQ(
+      parseDate("1880-04-13"), dateAdd(parseDate("1969-12-31"), kMinSmallint));
+}
+
+TEST_F(DateTimeFunctionsTest, dateAddTinyint) {
+  const auto dateAdd = [&](std::optional<int32_t> date,
+                           std::optional<int8_t> value) {
+    return evaluateDateFuncOnce<int32_t, int8_t>(
+        "date_add(c0, c1)", date, value);
+  };
+
+  // Check null behaviors
+  EXPECT_EQ(std::nullopt, dateAdd(std::nullopt, 1));
+  EXPECT_EQ(std::nullopt, dateAdd(parseDate("2019-02-28"), std::nullopt));
+  EXPECT_EQ(std::nullopt, dateAdd(std::nullopt, std::nullopt));
+
+  // Check simple tests.
```

Comment:
Would you add tests for "wrap around" behavior?

Reply:
Added the examples as tests, thanks.